tapgarden/custodian.go

```diff
@@ -420,7 +421,7 @@ func (c *Custodian) inspectWalletTx(walletTx *lndclient.Transaction) error {
 			// the proof will very likely fail. We should expect
 			// retrieval success before this delay.
 			select {
-			case <-time.After(defaultProofRetrievalDelay):
+			case <-time.After(c.cfg.ProofRetrievalDelay):
```

Comment:
Is this actually needed, now that we have proper retry?

Reply:
@Roasbeef Using this delay is one methodology which works. We could use another methodology without this delay.

I've documented this delay as follows:
> The number of seconds the custodian waits after identifying an asset transfer on-chain and before retrieving the corresponding proof.

My reasoning for this delay is that we shouldn't expect the first attempt at proof retrieval to work if we're immediately reacting to an on-chain tx. Because the sending node may not have had sufficient time to upload the proof to the proof courier. So, rather than try and expect a guaranteed failure, we delay so that our first attempt at retrieval might succeed.

I think we'll delay either way. Either with the backoff procedure after a first failed attempt. Or before hand, pre-emptively, to avoid failure on our first attempt.